core/src/main/java/org/infinispan/statetransfer/CommitManager.java

```diff
@@ -109,7 +140,8 @@ public final void stopTrack(Flag track) {
          return CompletableFutures.completedNull();
       }
       ByRef<CompletionStage<Void>> byRef = new ByRef<>(null);
-      tracker.compute(entry.getKey(), (o, discardPolicy) -> {
+      Map<Object, DiscardPolicy> entries =  tracker.computeIfAbsent(segment, s -> new ConcurrentHashMap<>());
```

Comment:
this should be `compute()`. The method below may not store anything in the `Map<Object, DiscardPolicy>` and you will end up with an empty map stored without need.

Reply:
Fixed it! The entries with empty map was the problem.